docs-gen/includes/quickstart/Dockerfile

```diff
@@ -1,10 +1,27 @@
-# TFX build
-FROM tensorflow/tfx:1.13.0
+FROM python:3.10.12 as builder
 
 RUN mkdir /data
 RUN wget https://raw.githubusercontent.com/tensorflow/tfx/master/tfx/examples/penguin/data/labelled/penguins_processed.csv -P /data
 
+RUN pip install poetry==1.7.1
```

Comment:
Can build a wheel file on the host and install that into the container instead similar to what we did for [the compiler](https://github.com/sky-uk/kfp-operator/blob/master/argo/kfp-compiler/Dockerfile#L4)?

It would mean we can use the host's python cache instead of the container's and there shouldn't be a need to compile on the target platform explicitly. It also means we probably don't need a virtual env?

Reply:
ok, will look into this today...

Reply:
As discussed, installing via wheels isn't any more efficient:
- pip dependency resolution still takes a very long time for `tfx` (docker build on a local machine did not complete for over 3 hours).
- cache improvements wouldn't help on CI

We need to create a virtual env if we want to take advantage of multistage builds. [This article](https://medium.com/@albertazzir/blazing-fast-python-docker-builds-with-poetry-a78a66f5aed0) is a fairly good approach.

We will raise a new issue to make the kfp-compiler image consistent with this approach.

Reply:
ok makes sense 👍